src/actions/tests/test_radio_interface.py

```diff
@@ -0,0 +1,233 @@
+""" Test aspects of RadioInterface """
+
+from os.path import join
+from actions import usrp
+
+
+class TestRadioInterface(object):
+
+    mock_scale_factor_dir = './actions/tests/mocks'
+    mock_bad_scale_factor_dir = \
+        './actions/tests/mocks/scale_factor_files_with_bad_formatting'
+
+    # Create the RadioInterface with the mock usrp_block
+    usrp.connect(use_mock_usrp=True)
+    if not usrp.is_available:
+        raise RuntimeError("Receiver is not available.")
+    rx = usrp.radio
+
+    # Ensure the usrp can recover from acquisition errors
+    def test_acquisition_errors(self):
+        """
+            The mock usrp_block will return "bad" data equal to
+            the number of times aquire_samples() has been called
+            until the reset_bad_acquisitions() has been called
+        """
+        self.rx.usrp.reset_bad_acquisitions()
+        max_retries = 5
+        for i in range(max_retries+2):
+            try:
+                self.rx.acquire_samples(1000, 1000, max_retries)
+            except RuntimeError:
+                # Check if this was the expected error
+                if i == max_retries+1:
+                    self.rx.usrp.reset_bad_acquisitions()
+                    return
+                else:
+                    raise RuntimeError(
+                        (
+                            "RadioInterface failed after {} retries.\r\n" +
+                            "Should have failed at {} retries."
+                        ).format(i-1, max_retries)
+                    )
+        # If no error was thrown, this is also an error
+        raise RuntimeError(
+            "RadioInterface did not throw expected error for\r\n" +
+            "failing to retrieve data {} times in a row.".format(max_retries+1)
+        )
+
+    def test_tune_result_parsing(self):
+        # Use a positive DSP frequency
+        f_lo = 1.0e9
+        f_dsp = 1.0e6
+        self.rx.tune_frequency(f_lo, f_dsp)
+        if not (f_lo == self.rx.lo_freq and
+                f_dsp == self.rx.dsp_freq):
+            raise RuntimeError(
+                (
+                    "Tune result parsing failed.\r\n" +
+                    "Set LO: {}, Parsed LO: {}\r\n" +
+                    "Set DSP: {}, Parsed DSP: {}"
+                ).format(
+                    f_lo, self.rx.lo_freq, f_dsp, self.rx.dsp_freq
+                )
+            )
```

Comment:
I'll refactor this into using a assert and also demonstrate how I like to deal with the 80-char limit without introducing a bunch of weird line breaks:

```python
def test_tune_result_parsing():
    f_lo = 1.0e9
    f_dsp = 1.0e6
    self.rx.tune_frequency(f_lo, f_dsp)
    msg = "Tune result parsing failed.\n"
    msg += "Set LO: {}, Parsed LO: {}\nSet DSP: {}, Parsed DSP: {}"
    msg = msg.format(f_lo, self.rx.lo_freq, f_dsp, self.rx.dsp_freq)
    lo_matches = f_lo == self.rx.lo_freq
    dsp_matches = f_dsp == self.rx.dsp_freq
    assert lo_matches and dsp_matches, msg
```

Reply:
Replaced similar instances with `assert (condition), msg`